### PR TITLE
Fixes issues with bad links in emails

### PR DESCRIPTION
**Pull request by datdamnzotz**

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -9,11 +9,19 @@ services:
     #     REPO: Orcpub
     #     BRANCH: develop
     environment:
+      # Domain or IP address of the site. example.com or 192.168.77.1
+      SERVER-DOMAIN:''
+      # Protocol used. Either 'http://' or 'https://'
+      SERVER-PROTOCOL: 'https://'
```
Comment (Demonsthere) [inline docker-compose.yaml:15]: I think this is a bit misleading, or could be worded differently. What do think about making it a `SSL_ENABLED=true/false/1/0` flag and casting it to Bool in code?

File: docker-compose.yaml [hunk 2]
```diff
@@ -9,11 +9,19 @@ services:
...
       PORT: 8890
       EMAIL_SERVER_URL: ''
       EMAIL_ACCESS_KEY: ''
       EMAIL_SECRET_KEY: ''
       EMAIL_SERVER_PORT: 587
+      # Errors 0 = Off, 1 = On
+      EMAIL_ERRORS: 0
```
Comment (Demonsthere) [inline docker-compose.yaml:22]: What do you think about `EMAIL_ERRORS_ENABLED' and also setting it to Bool?

File: docker-compose.yaml (hunk 1 above)
Comment (datdamnzotz) [reply docker-compose.yaml:15]: Fair enough. I'll update.

**Comment by datdamnzotz** [comment]
I'll refactor to bool

File: docker-compose.yaml (hunk 2 above)
Comment (MeanderingCode) [reply docker-compose.yaml:22]: You don't need this, just test if `EMAIL_ERRORS_TO` is set

File: src/clj/orcpub/email.clj [hunk 3]
```diff
@@ -30,15 +30,15 @@
    :host (environ/env :email-server-url)
    :port (Integer/parseInt (environ/env :email-server-port))})
 
-(defn send-verification-email [base-url {:keys [email username first-and-last-name]} verification-key]
+(defn send-verification-email [{:keys [email username first-and-last-name]} verification-key]
```
Comment (MeanderingCode) [inline src/clj/orcpub/email.clj:33]: You should not pull out `base-url`.

This is defining it as an argument to the function, which is called in the router `src/clj/orcpub/routes.clj` There, the request headers are being used to determine what the hostname is.

You should remove your environment variables, return this argument, and we should look into why it is getting  the wrong request headers. I suspect it is either not a problem, just an artifact of some setup, or due to the nginx reverse proxy config munging the request headers.

File: src/clj/orcpub/email.clj [hunk 4]
```diff
@@ -65,20 +65,21 @@
...
-  (postal/send-message (email-cfg)
-                       {:from "OrcPub Errors <no-reply@orcpub.com>"
-                        :to "redorc@orcpub.com"
-                        :subject "Exception"
-                        :body [{:type "text/plain"
-                                :content (let [writer (java.io.StringWriter.)]
-                                           (do (clojure.pprint/pprint (:request context) writer)
-                                               (clojure.pprint/pprint (or (ex-data exception) exception) writer)
-                                               (str writer)))}]}))
+  (if (= (environ/env :server-protocol) 1)
```
Comment (MeanderingCode) [inline src/clj/orcpub/email.clj:76]: Check here for `environ/env :email-errors-to` having a value.

**Review by MeanderingCode — COMMENTED** [review]
I know you closed this PR for rework, but I thought I would add my review for you before you get much further.

**Pull request closed without merging**